content/running_bookwyrm/cli.md

```diff
@@ -0,0 +1,143 @@
+
+## Managing a Bookwyrm instance
+
+### collectstatic
+
+Migrate static assets to either a Docker container or to an S3-compatible "bucket", depending on the context.
+
+### generate_preview_images
+
+Generate preview images for site, users, and books. This can take a while if you have a large database.
+
+### populate_streams
+
+Re-populates Redis streams (user feeds). You will usually need to run this after `update`.
+
+### populate_suggestions
+
+Populate suggested users for all users. You may want to run this manually to refresh suggestions.
+
+### restart_celery
+
+Restarts the `celery_worker` Docker container.
+
+### update
+
+When there are changes to the `production` branch, you can update your instance without downtime.
+
+This command `git pull`s the latest `production` branch updates, builds docker images if necessary, runs Django migrations, updates static files, and restarts all Docker containers.
+
+## Setting up S3 compatible storage
+
+By default, BookWyrm uses local storage for static assets (favicon, default avatar, etc.), and media (user avatars, book covers, etc.), but you can use an external storage service to serve these files. BookWyrm uses django-storages to handle external storage, such as S3-compatible services, Apache Libcloud or SFTP.
+
+See [External Storage](/external-storage.html) for more information.
+
+### copy_media_to_s3
+
+Migrate uploaded media from an existing Bookwrym installation to an S3-compatible "bucket".
+
+### set_cors_to_s3 _filename_
+
+Copy a CORS rules JSON file to your S3 bucket, where `filename` is the name of your JSON file (e.g. `./bw-dev set_cors_to_s3 cors.json`)
+
+## Development and testing
+
+### black
+
+BookWyrm uses the Black code formatter to keep the Python codebase consistent styled. Run `black` before committing your changes so the `pylint` task does not fail for your pull request and make you sad.
+
+### clean
+
+Remove all stopped Docker containers.
+
+Equivalent to:
+
+```shell
+docker-compose stop
+docker-compose rm -f
+```
+
+### makemessages
+
+Creates message files for all translation strings. After you have run `makemessages` you need to run `compilemessages` to compile the translations. See [Django's makemessages](https://docs.djangoproject.com/en/3.2/ref/django-admin/#makemessages).
+
+### compilemessages
+
+Compiles translation files. See [Django's compilemessages](https://docs.djangoproject.com/en/3.2/ref/django-admin/#compilemessages).
+
+### generate_thumbnails
+
+Generates thumbnail images for book covers.
+
+### pytest
+
+Run tests. This will not provide a coverage report if your tests fail: in that case use `test` for the coverage report.
```

Comment:
This actually should run a coverage report! I think the `test` command should be removed (at least, I never use it)

Reply:
oh I misread what you wrote; you're right that it doesn't produce a coverage report on test fail (I found it irritating to have to scroll past it to find the failed tests). I'm still going to remove the `test` command, because the tests should always be done with pytest and not with `coverage` or the default test library

Reply:
It's poorly written. If you remove `test` then the comment about coverage reports becomes irrelevant, I'll just get rid of it.